snuba/stateful_consumer/control_protocol.py

```diff
@@ -0,0 +1,172 @@
+from __future__ import annotations
+
+import jsonschema
+
+from dataclasses import dataclass
+from typing import Any, Mapping, Sequence
+from abc import ABC
+
+from snuba.snapshots import SnapshotId
+from snuba.snapshots.postgres_snapshot import Xid
+
+# TODO: Putting all messages in a single shema makes it hard to interpret errors
+# since the parser will always try all the options and it will rarely tell us which
+# is the actual error because it does not know which is the expected message.
+# We should try to split them into three schemas.
+CONTROL_MSG_SCHEMA = {
+    'anyOf': [
+        {"$ref": "#/definitions/snapshot-init"},
+        {"$ref": "#/definitions/snapshot-abort"},
+        {"$ref": "#/definitions/snapshot-loaded"},
+    ],
+    "definitions": {
+        "base": {
+            "type": "object",
+            "properties": {
+                "snapshot-id": {
+                    "type": "string"
+                }
+            },
+            "required": ["event", "snapshot-id"],
```

Comment:
Why is "event" part of this `required` list? It's not part of the property mapping, and seems like it's redundant with the definitions below?

Reply:
I think I just forgot it there. It took me a while to get this schema right, Likely I missed this one.